### DISCUSSION TITLE
Getting stuck when flashing Wifi Example on AI deck

### DISCUSSION THREAD

**Discussion by fairwel1** [bitcraze/discussions#1694] [Q&A]
Hi,
This problem has been bothering me for a long time, and I have tried  but still cannot solve it. My steps are as follows:

I just follow the Getting started with the AI deck [https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example](url) 

It looks like everything is ok when updateing Crazyflie and AIdeck firmware
![1](https://github.com/user-attachments/assets/260db02b-01fe-4091-8ec4-a7a73204550e)

And the OTA flashing stops at 3%. The crazyflie firstly restart after I enter cfloader command:
cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w radio://0/80/2M/E7E7E7E7E7
Then it will restart again about sevel seconds later.
![2](https://github.com/user-attachments/assets/a6964359-0911-4fb2-bae5-d4c68f7d6020)

I found my AI deck version is the older one(Rev.C1).I follow this step to flash the gap8 bootloader:[https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader](url) 
And the output seems no error :
![3](https://github.com/user-attachments/assets/c98d1466-0e9a-4bfb-a7f9-a750bc69f5f2)

But I retry the step flashing wifi example and still stuck at 3%. There seems no changing that the Crazyfile behaves like the second step . Just restarting twice and hangs.
![2](https://github.com/user-attachments/assets/adba62ea-e28e-4178-acf6-6cf73b542a24)

I find these similar problems mentioned before , but still couldn't slove it.
[https://github.com/orgs/bitcraze/discussions/173](url) 
[https://github.com/orgs/bitcraze/discussions/159](url)
[https://github.com/orgs/bitcraze/discussions/65](url)
[https://github.com/orgs/bitcraze/discussions/942](url)
[https://github.com/orgs/bitcraze/discussions/1573](url)
[https://github.com/orgs/bitcraze/discussions/1436](url)

So what's the issues? Can you give me some helps? 
Thanks!

**Comment by gemenerik** [comment]
Sorry to hear you are having problems flashing. Thanks for taking the time to investigate this and look into related discussions—it’s great that you’ve already gathered so much information.

Have you tried the resistor workaround mentioned in https://github.com/bitcraze/aideck-gap8-examples/issues/105? Adding a 4.5 kΩ resistor between VCOM and ground can help drain the capacitor properly, ensuring the system shuts off cleanly and allowing the flashing to proceed.

Also, are you using any additional decks with the AI Deck? If so, try removing them during the flashing process. Lastly, make sure the AI Deck is securely mounted, with all pins properly aligned and fully inserted.

**Reply by fairwel1** [reply]
I only attach AI Deck on Crazyflie and never use any other additional decks. So "The lighthouse deck seems to add capacitance on VCOM, causing the GAP8 to not reset correctly" metioned might not be the cause the failure of flashing on my device?And also it seems hard for me to add  4.5 kΩ resistor without tools available.

**Reply by gemenerik** [reply]
I'm afraid we've also seen this problem, though to a lesser degree, with just the AI deck. Adding the resistor between the pins should be relatively easy. You can even just use clamping wires on the pins, use the breakout deck or solder between the two. An alternative is to turn off the copter for a few minutes to allow the charge to drain, before turning it on again and flashing

**Reply by AvanishRanjit** [reply]
How did you did you solve it? Can you give some details!!!

**Reply by gemenerik** [reply]
Hi. We solved this by introducing a conditional boot delay a year ago. Are you running a recent firmware and still running into this problem? Then please open a new discussion with some details.

**Reply by AvanishRanjit** [reply]
I am running the latest firmware and also tried after dailing down the firmware version but it's not working are there any specific versions or stuff that I need to keep in mind.

**Comment by AvanishRanjit** [comment]
I did the entire process as mentioned in the bitcraze documentation https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/ . But I'm not able to flash the wifi bin file. I even tried first flashing the firmware of Gap8 and then trying to do the stuff. I did not try the capacitance thingy(which is mentioned above). My ai deck is revC does that require a different approach. Please help. The 3% thing is real.

**Reply by AvanishRanjit** [reply]
I was successful in flashing the firmware onto the gap8 using jtag but again the main flashing of the bin file for the wifi is not happening.

**Reply by gemenerik** [reply]
Are you saying you successfully flashed the gap8 bootloader with jtag before trying to flash the wifi example using the `cfloader` command? This is critical for it to work. Can you share the console output from when you flash the bootloader?

As a workaround you can also flash the wifi example directly using jtag:
- Clone https://github.com/bitcraze/aideck-gap8-examples
- Build and flash over jtag (make sure the debugger is connected to the gap8):
`docker run --rm -v ${PWD}:/module --device /dev/ttyUSB0 --privileged -P bitcraze/aideck tools/build/make-example examples/other/wifi-img-streamer flash`

**Reply by AvanishRanjit** [reply]
One question only now(because I have almost risked brain hemorrhage because I'm trying figure out it's wifi and stuff ), does the newer version of the esp32 skip the part where it needs to manually search for the wifi and is more automated in the newer 26.04 version where the connecting with wifi is more seamless, rather than older version where we had to like search for the wifi individually?

**Reply by gemenerik** [reply]
The WiFi setup hasn't changed. Not sure what you mean with more automated? It is possible to set up in station or AP mode.
